### PR TITLE
add a timeout to test-connection

### PR DESCRIPTION
**Pull request by jpsnover**
This replaces PR #2479

### CODE REVIEW

**Comment by JamesWTruher** [comment]
jeffrey and I spoke about my comments - the current timeout is 1000 ms, so this does not change our current behavior and provides for more flexibility

**Comment by JamesWTruher** [comment]
I'd love to see additional tests here - 
```
Describe "test-connection" -Tags @('Feature','Slow' ) {
     It "test-connection respects timeout" {
        $start = [datetime]::now
        Test-Connection -Computer _bad_computer_name_ -Timeout 10000 -ea silentlycontinue
        $end = [datetime]::now
        ($end - $start).TotalSeconds | should BeGreaterThan 10.0
     }
}
```
something along those lines

**Comment by andschwa** [comment]
Your committer email (private@jsnover.com) appears to be connected to the `jsnover` GitHub account, and not the `jpsnover` GitHub account you opened the PR from (so the commit doesn't "belong" to `jpsnover` as far as GitHub is concerned). Do you mean to use this email for you commits, and if so, is it possible to verify it with your current `jpsnover` GitHub account?

**Comment by jpsnover** [comment]
@andschwa I've made it so hat private@jsnover.com is the only account connected to my jpsnover github account.
let me know if I need to do anything else

File: src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs [hunk 1]
```diff
@@ -128,6 +128,14 @@ public class TestConnectionCommand : PSCmdlet
         public Int32 BufferSize { get; set; } = 32;
 
         /// <summary>
+        /// The following is the definition of the input parameter "TimeOut".
+        /// Time-out value in milliseconds. If a response is not received in this time, no response is assumed. The default is 1000 milliseconds.
+        /// </summary>
+        [Parameter]
+        [ValidateRange((int)1, Int32.MaxValue)]
```
Comment (lzybkr) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs]: I've never liked this use of `ValidateRange` - it seems like we should have `ValidatePositive` and `ValidateNonNegative` attributes.

**Pull request merged**